src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5815,6 +5815,20 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
                 HashSet<DiagnosticInfo> useSiteDiagnostics = null;
                 OverloadResolution.MethodInvocationOverloadResolution(methodGroup.Methods, methodGroup.TypeArguments, actualArguments, overloadResolutionResult, ref useSiteDiagnostics, isMethodGroupConversion, allowRefOmittedArguments);
                 diagnostics.Add(expression, useSiteDiagnostics);
+
+                if (overloadResolutionResult.Succeeded)
+                {
+                    var receiverParameter = overloadResolutionResult.BestResult.Member.Parameters.FirstOrDefault();
+                    Debug.Assert(receiverParameter != null);
+
+                    if (receiverParameter.RefKind == RefKind.Ref)
+                    {
+                        // If this was a ref extension method, the receiver argument must be checked for L-value constraints.
+                        // This helper method will also replace the symbol with a BadExpressionSymbol if it was invalid.
+                        actualArguments.Arguments.SetItem(0, CheckValue(left, BindValueKind.RefOrOut, diagnostics));
```

Comment:
>actualArguments.Arguments.SetItem(0, CheckValue(left, BindValueKind.RefOrOut, diagnostics)); [](start = 24, length = 92)

I am not sure if this is the right place to perform this check. For example, the argument might require conversion, which probably is going to invalidate this check. Also it is not clear what happens in a "Color-Color" case. It feels like BindInvocationExpressionContinued would be a better place.